gax-grpc/src/main/java/com/google/api/gax/grpc/ChannelPool.java

```diff
@@ -103,58 +97,66 @@ static ChannelPool create(int poolSize, ChannelFactory channelFactory) throws IO
    *
    * @param poolSize number of channels in the pool
    * @param channelFactory method to create the channels
-   * @param channelRefreshExecutorService periodically refreshes the channels; its life cycle will
-   *     be managed by ChannelPool
+   * @param executor used to schedule maintenance tasks like refresh channels and resizing the pool.
    * @return ChannelPool of refreshing channels
    */
   @VisibleForTesting
   static ChannelPool createRefreshing(
-      int poolSize,
-      ChannelFactory channelFactory,
-      ScheduledExecutorService channelRefreshExecutorService)
+      int poolSize, ChannelFactory channelFactory, ScheduledExecutorService executor)
       throws IOException {
-    return new ChannelPool(channelFactory, poolSize, channelRefreshExecutorService);
+    return new ChannelPool(
+        ChannelPoolSettings.staticallySized(poolSize)
+            .toBuilder()
+            .setPreemptiveReconnectEnabled(true)
+            .build(),
+        channelFactory,
+        executor);
   }
 
-  /**
-   * Factory method to create a refreshing channel pool
-   *
-   * @param poolSize number of channels in the pool
-   * @param channelFactory method to create the channels
-   * @return ChannelPool of refreshing channels
-   */
-  static ChannelPool createRefreshing(int poolSize, final ChannelFactory channelFactory)
+  static ChannelPool create(ChannelPoolSettings settings, ChannelFactory channelFactory)
       throws IOException {
-    return createRefreshing(
-        poolSize, channelFactory, Executors.newScheduledThreadPool(CHANNEL_REFRESH_EXECUTOR_SIZE));
+    return new ChannelPool(settings, channelFactory, Executors.newSingleThreadScheduledExecutor());
   }
 
   /**
    * Initializes the channel pool. Assumes that all channels have the same authority.
    *
+   * @param settings options for controling the ChannelPool sizing behavior
    * @param channelFactory method to create the channels
-   * @param poolSize number of channels in the pool
-   * @param channelRefreshExecutorService periodically refreshes the channels
+   * @param executor periodically refreshes the channels. Must be single threaded
```

Comment:
not anymore. It was meant to avoid contention between refreshing channels and resizing.
The code is now fully thread safe, so its no longer necessary